chainercv/links/model/vgg/vgg16.py

```diff
@@ -0,0 +1,299 @@
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+from chainer.initializers import constant
+from chainer.initializers import normal
+import chainer.links as L
+
+from chainercv.transforms import center_crop
+from chainercv.transforms import scale
+from chainercv.transforms import ten_crop
+
+from chainercv.utils import download_model
+
+
+# RGB order
+_imagenet_mean = np.array(
+    [123.68, 116.779, 103.939], dtype=np.float32)[:, np.newaxis, np.newaxis]
+
+
+class VGG16Layers(chainer.Chain):
+
+    """VGG16 Network for classification and feature extraction.
+
+    This model is a feature extraction link.
+    The network can choose to output features from set of all
+    intermediate and final features produced by the original architecture.
+
+    When :obj:`pretrained_model` is thepath of a pre-trained chainer model
+    serialized as a :obj:`.npz` file in the constructor, this chain model
+    automatically initializes all the parameters with it.
+    When a string in the prespecified set is provided, a pretrained model is
+    loaded from weights distributed on the Internet.
+    The list of pretrained models supported are as follows:
+
+    * :obj:`imagenet`: Loads weights trained with ImageNet and distributed \
+        at `Model Zoo \
+        <https://github.com/BVLC/caffe/wiki/Model-Zoo>`_.
+
+    Args:
+        pretrained_model (str): The destination of the pre-trained
+            chainer model serialized as a :obj:`.npz` file.
+            If this is one of the strings described
+            above, it automatically loads weights stored under a directory
+            :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/models/`,
+            where :obj:`$CHAINER_DATASET_ROOT` is set as
+            :obj:`$HOME/.chainer/dataset` unless you specify another value
+            by modifying the environment variable.
+        n_class (int): The dimension of the output of fc8.
+        feature (str or iterable of strings): The name of the feature to output
+            with :meth:`__call__` and :meth:`predict`.
+        initialW (callable): Initializer for the weights.
+        initial_bias (callable): Initializer for the biases.
+        mean (numpy.ndarray): A value to be subtracted from an image
+            in :meth:`_prepare`.
+        do_ten_crop (bool): If :obj:`True`, it averages results across
+            center, corners, and mirrors in :meth:`predict`. Otherwise, it uses
+            only the center. The default value is :obj:`False`.
+
+    """
+
+    _models = {
+        'imagenet': {
+            'n_class': 1000,
+            'url': 'https://github.com/yuyu2172/share-weights/releases/'
+            'download/0.0.3/vgg16_imagenet_convert_2017_06_15.npz'
+        }
+    }
+
+    def __init__(self, pretrained_model=None, n_class=None,
+                 features='prob', initialW=None, initial_bias=None,
+                 mean=_imagenet_mean, do_ten_crop=False):
+        if isinstance(features, (list, tuple)):
```

Comment:
How about checking whether `feature` is `str` or not? This is better because `iterable` is not only `list` and `tuple`.